### ISSUE TITLE
nodejs(all): npm install runs as root in the container, so can't write to disk

### ISSUE THREAD

**Issue by sam-github** [appsody/stacks#566]
**Describe the bug**

The `npm install` runs as root in the container, and since npm runs package-defined scripts, it has a protective mechanism to avoid running them as root, it drops its privileges to "nobody". This goes quite poorly when the package install has to write artifacts to disk, like every C++ addon does.

EDIT: by poorly, I mean impossible, the user nobody lacks file write permission, so c++ addons can't be compiled, their output can't be written

Reproduction:
```
appsody init nodejs-express
npm install modern-syslog
appsody run
cloud-node/ex-ext2 % appsody run
Running development environment...
Using local cache for image dev.local/nodejs-express:0.3.0                             
Running command: docker run --rm -p 3000:3000 -p 9229:9229 --name ex-ext2-dev -v /home/sam/w/cloud-node/ex-ext2/:/project/user-app -v ex-ext2-deps:/project/user-app/node_modu
les -v appsody-controller-0.3.1:/.appsody -t --entrypoint /.appsody/appsody-controller dev.local/nodejs-express:0.3.0 --mode=run
[Container] Running APPSODY_PREP command: npm install --prefix user-app && npm audit fix --prefix user-app
[Container] [..................] | rollbackFailedOptional: verb npm-session 42605430b3 

[Container] > modern-syslog@1.2.0 install /project/user-app/node_modules/modern-syslog 
[Container] > node-gyp rebuild
[Container]                                
[Container] gyp WARN EACCES user "undefined" does not have permission to access the dev dir "/root/.node-gyp/10.16.3"
[Container] gyp WARN EACCES attempting to reinstall using temporary dev dir "/project/user-app/node_modules/modern-syslog/.node-gyp"
[Container] gyp WARN install got an error, rolling back install                        
[Container] gyp WARN install got an error, rolling back install                        
[Container] gyp ERR! configure error 
[Container] gyp ERR! stack Error: EACCES: permission denied, mkdir '/project/user-app/node_modules/modern-syslog/.node-gyp'
[Container] gyp ERR! System Linux 5.3.0-23-generic                                     
[Container] gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
[Container] gyp ERR! cwd /project/user-app/node_modules/modern-syslog                  
[Container] gyp ERR! node -v v10.16.3
[Container] gyp ERR! node-gyp -v v3.8.0
[Container] gyp ERR! not ok 
[Container] npm ERR! code ELIFECYCLE
```

Some possible fixes for this:
- do `npm install --unsafe-perm`: this means when the install runs as root, it stays root. This might be OK since the install is occurring in a container anyhow, maybe exploitation of the container's root user is not a concern?
- run `npm install` as a non-root user, this might fix the other issue, where running as root is causing root-owned files to show up in the developer's account.

**Expected behavior**

npm install to suceed

**Actual behaviour**
npm install fails

**Environment Details (please complete the following information):**
 - OS: Linux (but since behaviour is in the container, should repro on other OS)

**If applicable please specify:**
 - CLI version: 
 - Stack you are using (including the version): nodejs-express, but should repro on any stack

**Screenshots**

**Additional context**

**Comment by sam-github** [comment]
Originally reported at https://github.com/appsody/stacks/issues/518#issuecomment-557698939, but since I found that bug is unrelated to npm (see https://github.com/appsody/stacks/issues/518#issuecomment-565157560).

**Comment by sam-github** [comment]
Possibly fixed by #589

**Comment by Kamran64** [comment]
Closing this issue: We'll use https://github.com/appsody/stacks/issues/518 to track the root owned node_modules and problems with running node stacks in general as non root user. (Since it looks like both issues are linked)
